src/js/craze-carousel.js

```diff
@@ -0,0 +1,126 @@
+
+
+global.jQuery = require('jquery');
+
+(function ( $ ) {
+ 
+    $.fn.crazeCarousel = function(options) {
+ 
+        var settings = $.extend({
+            autoRotate: true,
+            autoRotateMilisseconds: 4000,
+            transitionMilisseconds: 1000
+        }, options);
+
+        var self = this;
+        var children = $(this).children(".craze-carousel-item");
+        var currentIndex = 0;
+        var nextIndex = 1;
+
+        var addOnClickEventToLeftArrow = function() {
+            $(".craze-carousel_arrow-left-container").on("click", onClickedLeft);
+        }
+
+        var addOnClickEventToRightArrow = function() {
+            $(".craze-carousel_arrow-right-container").on("click", onClickedRight);
+        }
+
+        var onClickedLeft = function() {
+            settings.autoRotate = false;
+            rotateToPreviousImage();
+        }
+
+        var onClickedRight = function() {
+            settings.autoRotate = false;
+            rotateToNextImage();
+        }
+
+        var rotateToPreviousImage = function() {
+            let prevIndex = currentIndex - 1;
+            if (prevIndex < 0) nextIndex = children.length-1;
+            rotateToImageWithIndex(prevIndex);
+        }
+
+        var rotateToNextImage = function() {
+            let nextIndex = currentIndex + 1;
+            if (nextIndex > children.length-1) nextIndex = 0;
+            rotateToImageWithIndex(nextIndex);
+        }
+
+        var rotateToImageWithIndex = function(index) {
+            if (children.length > 1) {
+                for (var i = 0; i < children.length; i++) {
+                    $(children[i]).animate(
+                        {
+                            left: (-1 * (index) * 100) + '%'
+                        }, 
+                        { 
+                            duration: settings.transitionMilisseconds, 
+                            queue: false 
+                        }
+                    );
+                }
+
+                currentIndex = index;
+                setActiveCircleToIndex(currentIndex);
+            }
+        }
+
+        var autoRotate = function() {
+            if (settings.autoRotate) 
+                rotateToNextImage();
+        }
+
+        var setActiveCircleToIndex = function(index) {
+
+            let circles = $(self).children(".craze-carousel_circles-container").children(".craze-carousel_circle");
```

Comment:
Sometimes you are using `var`/ `let` to declare variables. Could you explain why?

Reply:
By definition, variables declared with `var` have function scope while variables declared with `let` have block scope. Even though both declarations are hoisted, if you try to access a `let` variable before its declaration you with get a `Reference Error`. Therefore, all of the `function` declarations and global variables in the `crazeCarousel` component should have use `var`.

Also, in the `crazeCarousel` component every single `let` variable declaration **could** have been a `var` declaration, since all of them are defined within a function of their own (their block scope is a function scope), but still it is a good practice not use `var` for a local auxiliary variable, since it might conflict with another of same name declared outside of that specific block.